packages/metals-vscode/src/extension.ts

```diff
@@ -1299,3 +1306,27 @@ function toggleBooleanWorkspaceSetting(setting: string) {
   const currentValues = configProperty?.workspaceValue ?? false;
   config.update(setting, !currentValues, ConfigurationTarget.Workspace);
 }
+
+function registerDebugEventListener(context: ExtensionContext) {
+  context.subscriptions.push(
+    debug.onDidReceiveDebugSessionCustomEvent((customEvent) => {
+      const t = customEvent.session ? customEvent.session.type : undefined;
+      if (t !== SCALA_LANGID) {
+        return;
+      }
+      if (customEvent.event === USER_NOTIFICATION_EVENT) {
+        handleUserNotification(customEvent);
+      }
```

Comment:
can `session` be null/undefined? Type suggests it can't. Anyway, I think following looks a bit clearer:
```suggestion
      const t = customEvent.session?.type;
      if (t === SCALA_LANGID && customEvent.event === USER_NOTIFICATION_EVENT) {
        handleUserNotification(customEvent);
      }
```